torch/_dynamo/variables/torch.py

```diff
@@ -895,7 +900,11 @@ def add_subgraph(name, gm):
                     next_name = candidate
 
             gm.__name__ = next_name
-            src = NNModuleSource(GetItemSource(self.source, next_name))
+            if self.source.guard_source().is_fsdp_module():
```

Comment:
what does 'add_subgraph' do generally? are there more than one subgraph per higher order operator?

Reply:
It is just a util that inserts the subgraph into the output graph module of Dynamo. Yes, there is no limit on how many subgraphs we can have. torch.cond, for example, has 2 - one for true branch and one for false.

In the case of activation checkpointing, there is only 1.